### PR TITLE
Replace all getOption/setOption usage with options

### PR DESCRIPTION
**Pull request by childrentime**
#3526

### CODE REVIEW

File: typings/xterm.d.ts [hunk 1]
```diff
@@ -1060,108 +1060,6 @@ declare module 'xterm' {
      */
     paste(data: string): void;
 
-    /**
-     * Retrieves an option's value from the terminal.
-     * @param key The option key.
-     * @deprecated Use `options` instead.
-     */
-    getOption(key: 'bellSound' | 'bellStyle' | 'cursorStyle' | 'fontFamily' | 'logLevel' | 'rendererType' | 'termName' | 'wordSeparator'): string;
```
Comment (Tyriar) [inline typings/xterm.d.ts:1068]: We'll have to bring this, the xterm-headless types and the public/Terminal.ts implementation back as we can only remove a deprecated feature in a major version.

**Review by Tyriar — CHANGES_REQUESTED** [review]
Looks good, just one comment

File: typings/xterm.d.ts (hunk 1 above)
Comment (childrentime) [reply typings/xterm.d.ts]: got it. done.

**Review by Tyriar — APPROVED** [review]
Thanks 👍

**Pull request merged**